platform/kubernetes/kubernetes.go

```diff
@@ -79,43 +76,35 @@ func isAddon(obj namespacedLabeled) bool {
 
 // --- /add ons
 
+type Applier interface {
+	Delete(logger log.Logger, def []byte) error
+	Create(logger log.Logger, def []byte) error
+	Apply(logger log.Logger, def []byte) error
```

Comment:
I cannot think of a concrete situation in which one is better than the other :S In lieu of that though, Paul suggests that declarations {Apply, Delete} may be easier to adapt to other platforms than operations {Create, Update, Delete}.